src/app/(steady)/steady/detail/[id]/page.tsx

```diff
@@ -1,78 +1,86 @@
 import Button, { buttonSize } from "@/components/_common/Button";
+import Dropdown from "@/components/_common/Dropdown";
 import Icon from "@/components/_common/Icon";
-import { AlertModal, UserModal } from "@/components/_common/Modal";
+import { AlertModal, InfoModal, UserModal } from "@/components/_common/Modal";
 import Tag from "@/components/_common/Tag";
+import { transformCreatedAt } from "@/utils/transformCreatedAt";
+import {
+  steadyCategoriesWithEmoji,
+  steadyExpectedPeriods,
+  steadyRecruitmentFields,
+  steadyRunningMethods,
+} from "@/constants/create-steady";
 
-const User = {
-  id: "11", // 유저 id
-  profileImageUrl:
-    "https://i.namu.wiki/i/w1bnPZXZOfmkgB1n2_1rQOIUhn-QAh4sNhDnLFAvlzmn7fSkV_XZMw-VFnG0ERPfifY9I0i54bQOMgHbPNBoyg.webp", // 프로필 이미지
-  nickname: "oort", // 닉네임
-  bio: "안녕하세요 전 윤하입니다", // 한 줄 소개
-  techInfo: ["react", "next", "typescript"], // 유저의 현재 관심 스택 정보
-  position: ["fe", "be"], // 유저의 현재 포지션 정보
-  //"formList": 유저가 작성한 폼 리스트
-  createdAt: "2023.10.29", // User 생성일
-};
-
-const SteadyPrimitive = {
-  id: "2", // 스테디 id,
-  masterId: "11", // 스테디 만든사람 id
-  title: "스테디를 만들자", // 스테디 제목
-  content: "스테디 모집합니다~", // 스테디 내용
-  type: "스터디", // 스테디 유형(스터디, 프로젝트)
-  createdAt: "2023.10.29", // 스테디 생성일(yyyy.mm.dd)
-  status: "모집", // 스테디 상태(모집/마감)
-  memberList: ["1", "3", "5", "7"], // 스테디에 참여한 유저id 목록
-  ended: true,
-  application: false, // 신청여부
-};
-
-const Announcement = {
-  title: "프론트 3명 구해요",
-  content: "프론트 같이 공부할 사람 구해요",
-  deadline: "2023.11.11", // 모집글 마감일
-  tags: ["프론트엔드", "리액트", "넥스트"], // 해쉬태그
-  createdAt: "2023.10.29", // 모집글 생성일
-  viewersNumber: 135, // 조회수
-  commentsNumber: 0, // 댓글 수
-  position: "프론트엔드", // 모집 분야
-  mode: "온라인", // 진행 방식(온/오프라인)
-  techStacks: [
-    {
-      id: "44",
-      name: "next", // techStackName
-      createdAt: Date, // Form 생성일
-    },
-    {
-      id: "55",
-      name: "react", // techStackName
-      createdAt: Date, // Form 생성일
-    },
-  ], // 기술스택
-  period: 3, // 예상기간
-  renewalTimes: 2, // 끌어올리기 남은 횟수
-};
 interface PageParams {
   id: string;
 }
 
 const SteadyDetailPage = ({ params }: { params: PageParams }) => {
-  const { data: steadyDetailsData } = useQuery({
-    queryKey: ["steadyDetails"],
-    queryFn: () => getSteadyDetails(params.id),
+  const { data: steadyDetailsData, refetch: steadyDetailsRefetch } =
+    useSuspenseQuery({
+      queryKey: ["steadyDetails"],
+      queryFn: () => getSteadyDetails(params.id),
+    });
+  const { data: steadyParticipantsData } = useSuspenseQuery({
+    queryKey: ["steadyParticipants"],
+    queryFn: () => getSteadyParticipants(params.id),
   });
   const router = useRouter();
-  console.log(params);
+  const { toast } = useToast();
+
+  const [isClient, setIsClient] = useState(false);
+
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  const handleClickPromoteBtn = async (steadyId: string) => {
+    if (steadyDetailsData.promotionCount === 0) {
```

Comment:
```suggestion
    if (!steadyDetailsData.promotionCount) {
```
물론 끌어올리기 횟수는 1씩 감소해서 0이 되겠지만, 예상치 못하게 음수값이 들어가게 되면 끌어올리기가 가능해지는 예외가 생길 수도 있을 것 같아요!

Reply:
```ts
steadyDetailsData.promotionCount <= 0
``` 
음수에 대해서는 다음과 같이 처리했습니당